### PR TITLE
Enable `git_stash_foreach`

### PR DESCRIPTION
**Pull request by johnhaley81**

### CODE REVIEW

**Comment by johnhaley81** [comment]
@tbranyen mind writing up some tests? :)

File: generate/input/callbacks.json [hunk 1]
```diff
@@ -424,7 +424,7 @@
       },
       {
         "name": "stash_id",
-        "cType": "const int *"
+        "cType": "const git_oid *"
```
Comment (tbranyen) [inline generate/input/callbacks.json]: omg....

Comment (johnhaley81) [inline generate/input/callbacks.json]: Don't trust the docs :)

**Comment by johnhaley81** [comment]
@elefthei has offered to write up some tests for this.

**Pull request merged**